magda-gateway/src/createBaseProxy.ts

```diff
@@ -1,4 +1,15 @@
 import * as httpProxy from "http-proxy";
+
+import * as URI from "urijs";
+import {
+    magdaAdminPortalName,
+    MAGDA_TENANT_ID_HEADER,
+    MAGDA_ADMIN_PORTAL_ID,
+    multiTenantsMode
+} from "./index";
+
+import { tenantsTable } from "./buildApp";
```

Comment:
Similarly here, a separate module should be used to manage the tenants table. There's probably a cycle of imports as a result of adding this, i.e. `buildApp` depends indirectly on `createBaseProxy`, and `createBaseProxy` also depends on `buildApp`. I'm sort of surprised it's not causing more trouble.

Reply:
fixed.